### PR TITLE
Record data

### PR DESCRIPTION
**Pull request by Narukodo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,38 +12,31 @@ Speech data acquired from [Kaggle's speech accent archive](https://www.kaggle.co
 - Slow speech: hungarian8.wav
 - Thick accent: bambara4.wav
 - Normal speech: serbian9.wav
+An exerpt of Trump speaking on COVID-19 was used a real life example
```
Comment (RodericDay) [inline README.md]: exerpt -> excerpt

File: README.md [hunk 2]
```diff
@@ -13,38 +12,31 @@ Speech data acquired from [Kaggle's speech accent archive](https://www.kaggle.co
...
+An exposition of the book of Philippians was used as a lengthy example (approx. 9 minutes)
 
 ## Engines
-- IBM Watson
-- Dialogflow
 - CMUSphinx
 - Google Speech Recognition
-- Kaldi
-- Houndify
```
Comment (RodericDay) [inline README.md:23]: oh, I see.

You could mention you tried some of the others but they were difficult to implement, just as an "I tried them btw not news to me"

File: experiment/experiment.py [hunk 3]
```diff
@@ -3,42 +3,39 @@
...
+        sound_bytes = getattr(wav_info, field)
         start_time = datetime.datetime.now()
         try:
             guess = speech_to_text_fn(sound_bytes)  # will take in a file
         except sr.UnknownValueError:
             return DataPoint(filename, -1, -1, -1)
         end_time = datetime.datetime.now()
         latency = start_time - end_time
-        return DataPoint(filename, latency.seconds, latency.microseconds * 1000 , self.calculate_accuracy(guess, label))
+        return DataPoint(filename, self.calculate_accuracy(guess, label))
```
Comment (RodericDay) [inline experiment/experiment.py:38]: love this breaking things into descriptive functions

File: google_cloud.py [hunk 4]
```diff
@@ -30,7 +26,10 @@ def sample_long_running_recognize(storage_uri):
     }
     audio = {"uri": storage_uri}
 
-    operation = client.long_running_recognize(config, audio)
+    try:
+        operation = client.long_running_recognize(config, audio)
+    except:
```
Comment (RodericDay) [inline google_cloud.py]: naked excepts are not a good idea generally. is there any possibility of narrowing it down a bit, to expected errors?

File: sounds.py [hunk 5]
```diff
@@ -1,22 +1,48 @@
 from collections import namedtuple
+from dataclasses import dataclass
+from dotenv import load_dotenv
+from google.cloud import storage
 import speech_recognition as sr
-from pathlib import Path 
+from speech_recognition import AudioData
 import os
+from pathlib import Path 
 import re
```
Comment (RodericDay) [inline sounds.py:6]: a little flake8 / PEP8 pass could help make these import sections a bit cleaner

File: sounds.py [hunk 6]
```diff
@@ -1,22 +1,48 @@
...
+r = sr.Recognizer()
 
-sound_samples_folder = Path().cwd() / 'sounds/samples'
-# indicate which text to compare speech to
-speech_labels = [folder_name for folder_name in os.listdir(sound_samples_folder) if os.path.isdir(sound_samples_folder / folder_name)]
-wav_filenames = []
-for speech_label in speech_labels:
-    curfolder_wavfiles = [(speech_label, wav_filename) for wav_filename in os.listdir(sound_samples_folder / speech_label) if re.match(r'^.+\.wav$', wav_filename)]
-    wav_filenames = wav_filenames + curfolder_wavfiles
+@dataclass
```
Comment (RodericDay) [inline sounds.py:11]: amazing. you could tell people about why you used this and why, it's a cool modern Python feature that I don't think a lot of people know about

**Review by RodericDay — CHANGES_REQUESTED** [review]

File: sounds.py (hunk 5 above)
Comment (Narukodo) [reply sounds.py:6]: flake8 seems to be okay with this actually

**Pull request merged**